compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -72,16 +72,16 @@ pub(super) struct Monomorphizer<'interner> {
     /// duplicated during monomorphization. Doing so would allow them to be used polymorphically
     /// but would also cause them to be re-evaluated which is a performance trap that would
     /// confuse users.
-    locals: HashMap<node_interner::DefinitionId, LocalId>,
+    pub locals: HashMap<node_interner::DefinitionId, LocalId>,
```

Comment:
I'd consider adding read-only accessors to expose these without the ability to modify them, so that `Monomorphizer` can properly maintain its own data and invariants, while allowing 3rd parties to query it.